src/Plugin/Commerce/PriceRuleCalculation/AttributePrice.php

```diff
@@ -0,0 +1,227 @@
+   *   the attributes used as value.
+   */
+  public function getAttributesPriceField() {
+    $price_fields = [];
+    $attribute_storage = $this->entityTypeManager->getStorage('commerce_product_attribute');
+    $attributes = $attribute_storage->loadMultiple();
+    foreach ($attributes as $attribute) {
+      $fields = \Drupal::service('entity_field.manager')->getFieldDefinitions('commerce_product_attribute_value', $attribute->id());
+      foreach ($fields as $field) {
+        $type = $field->getType();
+        if ($type == "commerce_price") {
+          $price_fields[$attribute->id()]['commerce_price_fields'][$field->get('field_name')] = $field->get('label');
+        }
+      }
+    }
+    return $price_fields;
+  }
+
+  /**
+   * Provides the total price of attribues attached to a variation.
+   *
+   * @param Drupal\commerce_product\Entity\ProductVariation $variation
+   *   The commerce product variation.
+   * @param array $attribute_field_map
+   *   The attribute price field mappings.
+   *
+   * @return Drupal\commerce_price\Price
+   *   Sum of all attribute values associated with a variation.
+   */
+  public function getAttributeTotalPrice(ProductVariation $variation, array $attribute_field_map) {
+    $attribute_total_price = new Price((string) 0, $variation->get('price')->first()->toPrice()->getCurrencyCode());
+    $attribute_storage = $this->entityTypeManager->getStorage('commerce_product_attribute');
+    $attributes = $attribute_storage->loadMultiple();
+    if (empty($attributes)) {
+      return $attribute_total_price;
+    }
+    foreach ($attributes as $attribute) {
+      // The attribute field name of variation type will always be in the
+      // format of `attrubute_attribute_machine_name`.
+      $attribute_field_name = 'attribute_' . $attribute->id();
+      $attribute_value = $variation->get($attribute_field_name)->entity;
+      if (!empty($attribute_value) && $attribute_value->hasField($attribute_field_map[$attribute->id()]['field'])) {
+        $attribute_price = $attribute_value->get($attribute_field_map[$attribute->id()]['field'])->first()->toPrice();
+        $attribute_total_price = $attribute_total_price->add($attribute_price);
+      }
+    }
+    return $attribute_total_price;
+  }
+
+  /**
+   * Recalculates the variation price.
+   *
+   * @param Drupal\commerce_product\Entity\ProductVariation $variation
+   *   The commerce product variation.
+   * @param array $attribute_field_map
+   *   The attribute price field mappings.
+   *
+   * @return Drupal\commerce_price\Price
+   *   The recalculated variation price.
+   */
+  public function recalculateVariationPrice(ProductVariation $variation, array $attribute_field_map) {
+    $base_price = $variation->get('price')->first()->toPrice();
+    $attribute_price = $this->getAttributeTotalPrice($variation, $attribute_field_map);
+    $price = $base_price->add($attribute_price);
+    return $price;
```

Comment:
**CHANGE REQUEST**

*Performance*

No need to save the result in a variable just so that you return it.

```
    $base_price = $variation->get('price')->first()->toPrice();
    $attribute_price = $this->getAttributeTotalPrice($variation, $attribute_field_map);

    return $base_price->add($attribute_price);
```

Reply:
done